backend/sirius-components-collaborative-diagrams/src/main/java/org/eclipse/sirius/components/collaborative/diagrams/handlers/GetConnectorToolsEventHandler.java

```diff
@@ -0,0 +1,149 @@
+import reactor.core.publisher.Sinks.Many;
+import reactor.core.publisher.Sinks.One;
+
+/**
+ * Event handler for "get connector tools" query.
+ *
+ * @author nvannier
+ */
+public class GetConnectorToolsEventHandler implements IDiagramEventHandler {
+
+    private final IRepresentationDescriptionSearchService representationDescriptionSearchService;
+
+    private final IDiagramQueryService diagramQueryService;
+
+    private final IConnectorToolsProviderRegistry connectorToolsProviderRegistry;
+
+    private final IObjectService objectService;
+
+    private final ICollaborativeMessageService messageService;
+
+    private final Counter counter;
+
+    public GetConnectorToolsEventHandler(IRepresentationDescriptionSearchService representationDescriptionSearchService, IDiagramQueryService diagramQueryService,
+            IConnectorToolsProviderRegistry connectorToolsProviderRegistry, IObjectService objectService, ICollaborativeMessageService messageService, MeterRegistry meterRegistry) {
+        this.representationDescriptionSearchService = Objects.requireNonNull(representationDescriptionSearchService);
+        this.diagramQueryService = Objects.requireNonNull(diagramQueryService);
+        this.connectorToolsProviderRegistry = Objects.requireNonNull(connectorToolsProviderRegistry);
+        this.objectService = Objects.requireNonNull(objectService);
+        this.messageService = Objects.requireNonNull(messageService);
+
+        // @formatter:off
+        this.counter = Counter.builder(Monitoring.EVENT_HANDLER)
+                .tag(Monitoring.NAME, this.getClass().getSimpleName())
+                .register(meterRegistry);
+        // @formatter:on
+    }
+
+    @Override
+    public boolean canHandle(IDiagramInput diagramInput) {
+        return diagramInput instanceof GetConnectorToolsInput;
+    }
+
+    @Override
+    public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, IDiagramContext diagramContext, IDiagramInput diagramInput) {
+        this.counter.increment();
+
+        ChangeDescription changeDescription = new ChangeDescription(ChangeKind.NOTHING, editingContext.getId(), diagramInput);
+        IPayload payload = null;
+
+        if (diagramInput instanceof GetConnectorToolsInput) {
+            GetConnectorToolsInput connectorToolsInput = (GetConnectorToolsInput) diagramInput;
+            Diagram diagram = diagramContext.getDiagram();
+            //@formatter:off
+            var diagramDescription = this.representationDescriptionSearchService.findById(editingContext, diagram.getDescriptionId())
+                    .filter(DiagramDescription.class::isInstance)
+                    .map(DiagramDescription.class::cast);
+            //@formatter:on
+
+            if (diagramDescription.isPresent()) {
+                var connectorToolsProvider = this.connectorToolsProviderRegistry.get(diagramDescription.get());
```

Comment:
Multiple providers should be able to contribute at the same time if they all support the same diagram description.